bridge/src/main/java/io/stargate/bridge/service/QueryHandler.java

```diff
@@ -154,14 +154,21 @@ private ByteBuffer getComparableBytesFromRow(
   }
 
   private ByteBuffer getPagingStateFromRow(
-      ByteBuffer resultSetPagingState, Row row, boolean lastInPage) {
+      ByteBuffer resultSetPagingState,
+      Row row,
+      QueryOuterClass.ResumeMode resumeMode,
+      boolean lastInPage) {
     if (lastInPage && resultSetPagingState == null) {
       return EXHAUSTED_PAGE_STATE;
     }
 
+    PagingPosition.ResumeMode internalResumeMode = PagingPosition.ResumeMode.NEXT_ROW;
+    if (resumeMode != null && resumeMode.getValueDescriptor().getNumber() == 1) {
```

Comment:
Is there a better way to match the enum's value to a PagingPosition.ResumeMode? Should I use the name of the enum instead?

Reply:
Use `QueryOuterClass.ResumeMode.NEXT_PARTITION`?

```diff
diff --git a/bridge/src/main/java/io/stargate/bridge/service/QueryHandler.java b/bridge/src/main/java/io/stargate/bridge/service/QueryHandler.java
index 21db6e2b6..47dadf815 100644
--- a/bridge/src/main/java/io/stargate/bridge/service/QueryHandler.java
+++ b/bridge/src/main/java/io/stargate/bridge/service/QueryHandler.java
@@ -26,6 +26,7 @@ import io.stargate.bridge.proto.QueryOuterClass.SchemaChange;
 import io.stargate.db.ClientInfo;
 import io.stargate.db.ImmutableParameters;
 import io.stargate.db.PagingPosition;
+import io.stargate.db.PagingPosition.ResumeMode;
 import io.stargate.db.Parameters;
 import io.stargate.db.Persistence;
 import io.stargate.db.Persistence.Connection;
@@ -40,6 +41,7 @@ import java.nio.ByteBuffer;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
@@ -163,7 +165,7 @@ public class QueryHandler extends MessageHandler<Query, Prepared> {
     }
 
     PagingPosition.ResumeMode internalResumeMode = PagingPosition.ResumeMode.NEXT_ROW;
-    if (resumeMode != null && resumeMode.getValueDescriptor().getNumber() == 1) {
+    if (Objects.equals(resumeMode, QueryOuterClass.ResumeMode.NEXT_PARTITION)) {
       internalResumeMode = PagingPosition.ResumeMode.NEXT_PARTITION;
     }
```